apps/wallet/src/ui/app/shared/faucet/useFaucetMutation.ts

```diff
@@ -26,15 +26,48 @@ export function useFaucetMutation(options?: UseFaucetMutationOptions) {
 				throw new Error('Failed, faucet host not found.');
 			}
 
-			const { error, transferredGasObjects } = await requestSuiFromFaucetV0({
+			const { error, task: taskId } = await requestSuiFromFaucetV1({
 				recipient: addressToTopUp,
 				host: options.host,
 			});
+			if (!taskId) {
+				throw new Error('Failed, task id not found.');
+			}
+			// Initialize a variable to track possible faucet request status errors
+			let faucetStatusError: string | null = null;
+			let faucetStatusTransferObject: number | null = null;
+			if (taskId) {
+				// Continuously check the status until it's no longer 'INPROGRESS'
+				let currentStatus = 'INPROGRESS';
+				while (currentStatus === 'INPROGRESS') {
+					const { status, error } = await getFaucetRequestStatus({
+						host: options.host,
+						taskId,
+					});
+
+					if (status?.status === 'SUCCEEDED') {
+						faucetStatusTransferObject = status.transferredGasObjects?.reduce(
```

Comment:
a sample response is
```
{
    "status": {
        "status": "SUCCEEDED",
        "transferred_gas_objects": {
            "sent": [
                {
                    "amount": 10000000000,
                    "id": "0x700e75d24bb7d4cbecbe7d54c8b026d32d636a8644f3afae12ce5cf7cb05e722",
                    "transferTxDigest": "HzaLNMAyCLn4pCZP7z1WTMcvLbQLJr8itKRXu6R7DJzC"
                }
            ]
        }
    },
    "error": null
}
```

So this should be `status.transferred_gas_objects.sent...`?